src/components/Switch.js

```diff
@@ -16,59 +16,39 @@ const propTypes = {
     accessibilityLabel: PropTypes.string.isRequired,
 };
 
-const PressableWithFeedback = Pressables.PressableWithFeedback;
-class Switch extends Component {
-    constructor(props) {
-        super(props);
-        this.offPosition = 0;
-        this.onPosition = 20;
-        this.offsetX = new Animated.Value(props.isOn ? this.onPosition : this.offPosition);
-
-        this.toggleSwitch = this.toggleSwitch.bind(this);
-        this.toggleAction = this.toggleAction.bind(this);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.isOn === this.props.isOn) {
-            return;
-        }
+const OFFSET_X = {
+    OFF: 0,
+    ON: 20,
+};
 
-        this.toggleSwitch();
-    }
+function Switch(props) {
+    const offsetX = useRef(new Animated.Value(props.isOn ? OFFSET_X.ON : OFFSET_X.OFF)).current;
```

Comment:
```suggestion
    const offsetX = useRef(new Animated.Value(props.isOn ? OFFSET_X.ON : OFFSET_X.OFF));
```

Is there any reason to define this as the `current` ref value? If not please apply the suggested change, this is helpful to avoid the `useEffect` extra dependencies (or lint disable rules)